### PR TITLE
Add example usage in Android project

### PR DESCRIPTION
**Pull request by KeithYokoma**
## Description

Now we have an example project for Android app using RemoteDataK Android, using Android Architecture Component ViewModel + SavedStateHandle for keeping state beyond Configuration Changes / Don't Keep Activities.

## Screenshot

device-2020-04-16-130611

### CODE REVIEW

File: remotedata_android_sample/src/main/java/com/mercari/remotedata/android/sample/ui/main/MainViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+@Parcelize
+class IllegalState : ErrorKind
+@Parcelize
+class UnknownError : ErrorKind
+
+private fun Throwable.toErrorKind(): ErrorKind = when (this) {
+    is IllegalStateException -> IllegalState()
+    else -> UnknownError()
+}
```
Comment (yhanada) [inline remotedata_android_sample/src/main/java/com/mercari/remotedata/android/sample/ui/main/MainViewModel.kt:73]: [Just a comment]
This extension could be used in other products too 😍

**Review by yhanada — APPROVED** [review]
Thanks for the easy-to-understand sample 👍

**Comment by kittinunf** [comment]
@KeithYokoma Do you think we also have an example for folks who don't use `LiveData<T>`, just RxJava's `Observable`?

**Comment by KeithYokoma** [comment]
Sure thing!

**Pull request merged**